yb-voyager/src/srcdb/pg_dump_extract_schema.go

```diff
@@ -29,10 +29,9 @@ func pgdumpExtractSchema(source *Source, connectionUri string, exportDir string)
 	if !source.CommentsOnObjects {
 		pgDumpArgs = fmt.Sprintf(`%s --no-comments`, pgDumpArgs)
 	}
+	os.Setenv("PGPASSWORD", source.Password)
```

Comment:
You can just add `defer os.Unsetenv("PGPASSWORD")` immediately after this call. That should take care of clearing the env var.